### DISCUSSION TITLE
Retrieving data of a table starting from a join table (many-to-many)

### DISCUSSION THREAD

**Discussion by noook** [supabase/supabase#13348] [Questions]
I'm very new to Supabase and can't find any solution to my problem:
I have 2 tables: `public.users` which is a view of `auth.users` and `leaderboards`. I need to fetch all leaderboards a user has access to. So here is my SQL query:
```sql
SELECT leaderboard.* from leaderboard_users
LEFT JOIN public.users ON leaderboard_users.user_id = public.users.id
LEFT JOIN leaderboard ON leaderboard_users.leaderboard_id = leaderboard.id
```

The result is exactly what I need in SQL, however it returns me this payload:
```json
[
  {
    "leaderboard": {
      "id": "0d8789a2-f152-4891-8996-ca71e4a4ccdc",
      "name": "Leaderboard 1",
      "owner": "8f848836-f43c-45c6-abce-255eed8706a8",
      "created_at": "2023-03-31T15:59:02.054744+00:00"
    }
  }
]
```

Is there a way to build a query in a way that it returns this ?
```json
[
  {
    "id": "0d8789a2-f152-4891-8996-ca71e4a4ccdc",
    "name": "Leaderboard 1",
    "owner": "8f848836-f43c-45c6-abce-255eed8706a8",
    "created_at": "2023-03-31T15:59:02.054744+00:00"
  }
]
```

This is the request I have so far:
```ts
const { data } = await supabase
  .from('leaderboard')
  .select('*, leaderboard_users(*)');
```

This returns
```json
[
  {
    "id": "0d8789a2-f152-4891-8996-ca71e4a4ccdc",
    "name": "Leaderboard 1",
    "owner": "8f848836-f43c-45c6-abce-255eed8706a8",
    "created_at": "2023-03-31T15:59:02.054744+00:00",
    "leaderboard_users": [
      {
        "id": 1,
        "leaderboard_id": "0d8789a2-f152-4891-8996-ca71e4a4ccdc",
        "user_id": "8f848836-f43c-45c6-abce-255eed8706a8",
        "joined_at": "2023-03-31T15:59:39.156727+00:00"
      }
    ]
  },
  {
    "id": "5828e478-04fc-4e7f-a659-3c27615ac212",
    "name": "Leaderboard 2",
    "owner": "79e4925c-fd58-4c38-8aac-0684e89ae318",
    "created_at": "2023-03-31T16:47:39.520821+00:00",
    "leaderboard_users": [
      {
        "id": 2,
        "leaderboard_id": "5828e478-04fc-4e7f-a659-3c27615ac212",
        "user_id": "79e4925c-fd58-4c38-8aac-0684e89ae318",
        "joined_at": "2023-03-31T16:47:54.808349+00:00"
      }
    ]
  }
]
```

However, I want to filter the leaderboard in a way that I can only see those where a user with a given ID belongs to.

So I tried filtering with `.eq()`:
```ts
const { data } = await supabase
  .from('leaderboard')
  .select('*, leaderboard_users(*)')
  .eq('leaderboard_users.user_id', user.id);
```

But it only filters out the users from the `leaderboard_users`:
```json
[
  {
    "id": "0d8789a2-f152-4891-8996-ca71e4a4ccdc",
    "name": "Leaderboard 1",
    "owner": "8f848836-f43c-45c6-abce-255eed8706a8",
    "created_at": "2023-03-31T15:59:02.054744+00:00",
    "leaderboard_users": [
      {
        "id": 1,
        "leaderboard_id": "0d8789a2-f152-4891-8996-ca71e4a4ccdc",
        "user_id": "8f848836-f43c-45c6-abce-255eed8706a8",
        "joined_at": "2023-03-31T15:59:39.156727+00:00"
      }
    ]
  },
  {
    "id": "5828e478-04fc-4e7f-a659-3c27615ac212",
    "name": "Leaderboard 2",
    "owner": "79e4925c-fd58-4c38-8aac-0684e89ae318",
    "created_at": "2023-03-31T16:47:39.520821+00:00",
    "leaderboard_users": []
  }
]
```

In this situation, if I am the user with the id **2**, I want to see only the **`Leaderboard 1`**. Also, I don't need to have the `leaderboard_users` key, I just need to filter on this, but I don't want it at the end

**Comment by steve-chavez** [comment]
How are you running that SQL? Through RPC? If so, can you share the function definition?

**Reply by noook** [reply]
I'm trying to convert that SQL query into a `.select` query with Supabase's syntax and this is what I'm trying to figure out

**Reply by noook** [reply]
I updated the question to give you more context, and how I tried to achieve it.

**Reply by GaryAustin1** [reply]
Try `.select('*, leaderboard_users!inner(*)')`

**Reply by noook** [reply]
Works ! However is there a way not to get the `leaderboard_users` property ? It's not relevant data and the user should not be aware of it

**Reply by GaryAustin1** [reply]
Not quite yet, unless you are developing locally.
https://supabase.com/blog/postgrest-11-prerelease
See `spreading related tables`.
Post process is way at the moment, all though Steve is main PostgREST guy, so will probably chime in on status of that for hosted.

**Reply by steve-chavez** [reply]
On next week postgREST 11 will be made available for hosted.
